### ISSUE TITLE
Don't suggest String.Concat for so many cases of +

### ISSUE THREAD

**Issue by ljw1004** [Microsoft/RoslynClrHeapAllocationAnalyzer#26]
``` csharp
string s = "hello" + o.ToString() + "world";
```

The C# compiler translates this into a call to String.Concat. You can see that in the IL it emits, and the source code for the LocalRewriter_StringConcat.cs function is [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs,e2a9eb71cf827d41).

This analyzer currently suggests to turn the above C# code into a call to `String.Concat`. I think it shouldn't: this warning is just noise.

**Comment by mjsabby** [comment]
Yes I agree. This one was shoehorned anyway, so should be relatively easy to remove. PR welcome.
